main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -995,6 +1002,20 @@ object Weeder2 {
       }
     }
 
+    /**
+     * This method is the same as visitArguments but for invokeMethod2. It does not consider named arguments
+     * as they are not allowed and it doesn't add unit arguments for empty arguments.
+     */
+    private def visitMethodArguments(tree: Tree): Validation[List[Expr], CompilationMessage] = {
+      mapN(
```

Comment:
This can probably just be simplified to `traverse(pickAll(TreeKind.Argument, tree))(pickExpr)`